core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGenesetHierarchy.java

```diff
@@ -75,11 +75,6 @@ public void run() {
 
 			// Check if geneset_hierarchy_node already filled
 			boolean emptyDatabase = !DaoGenesetHierarchyNode.checkGenesetHierarchy();
-			if (emptyDatabase) {
```

Comment:
Would moving the `emptyDatabase` boolean down to the line on which it is needed break things? And if so, could you add a comment on how `importData()` (which apparently validates something) does interact with `checkGenesetHierarchy()`?